keysign/gpgmeh.py

```diff
@@ -471,3 +474,34 @@ def sign_keydata_and_encrypt(keydata, error_cb=None, homedir=None):
                                                always_trust=True,
                                                sign=False)
                 yield (UID.from_gpgme(uid), ciphertext)
+
+
+def import_signature(encrypted_sig, homedir=None):
+    # ctx = gpg.Context()
+    ctx = DirectoryContext(homedir)
+    signature = ctx.decrypt(encrypted_sig)
+
```

Comment:
But we don't know yet whether this is indeed a signature, right?
It's not a problem for now, but I feel that we want to know that we do indeed have decrypted a signature. Bonus points if its a signature for one of our keys.

I think the best way to go about it is to use a TemporaryContext, import there, check gpgme's import result for having imported a new signature rather than a new key, and then continue.

Reply:
Well, if decrypt succeed we are sure that it is a signature.
From the doc of decrypt:
"Decrypt the given ciphertext and verify any signatures. If
VERIFY is an iterable of keys, the ciphertext must be signed
by all those keys, otherwise an error is raised."

And in fact if you try to use a file that is not a valid signature an error will be immediately raised:
```
keysign.send (INFO): Received file: /home/ryuzaki/DefinitelyNotASig.txt
keysign.send (ERROR): Could not import signatures
Traceback (most recent call last):
  File "/home/ryuzaki/Github/gnome-keysign/keysign/send.py", line 136, in on_drag_data_received
    gpgmeh.import_signature(signature)
  File "/home/ryuzaki/Github/gnome-keysign/keysign/gpgmeh.py", line 482, in import_signature
    signature = ctx.decrypt(encrypted_sig)
  File "/usr/lib/python3.7/site-packages/gpg/core.py", line 409, in decrypt
    raise e
  File "/usr/lib/python3.7/site-packages/gpg/core.py", line 398, in decrypt
    self.op_decrypt_verify(ciphertext, plaintext)
  File "/usr/lib/python3.7/site-packages/gpg/core.py", line 158, in wrapper
    return _funcwrap(self, *args)
  File "/usr/lib/python3.7/site-packages/gpg/core.py", line 141, in _funcwrap
    return errorcheck(result, name)
  File "/usr/lib/python3.7/site-packages/gpg/errors.py", line 129, in errorcheck
    raise GPGMEError(retval, extradata)
gpg.errors.GPGMEError: gpgme_op_decrypt_verify: GPGME: No data
keysign.send (INFO): Signature import error
```

Reply:
With the commit 609d285 now we check if we are really importing a signature or a new key.